openfisca_core/entities.py

```diff
@@ -364,6 +381,16 @@ def members_position(self):
     def members_role(self, members_role):
         self._members_role = members_role
 
+    @property
+    def roles_count(self):
+        if self._roles_count is None:
+            self._roles_count = self.members_legacy_role.max() + 1
+        return self._roles_count
+
+    @roles_count.setter
+    def roles_count(self, roles_count):
```

Comment:
This is Python [property](https://www.programiz.com/python-programming/property).

The goal of this change is that if you call `entity.roles_count`, and that this attribute has not been set yet, then it will be deducted from `entity. members_legacy_role`

FI the attribute `roles_count ` is only needed for the "old" syntax.

Reply:
Oooh got it !